src/esp/core/ManagedContainer.h

```diff
@@ -20,8 +20,11 @@ namespace core {
  * @tparam T the type of managed object a particular specialization of
  * this class works with.  Must inherit from @ref
  * esp::core::AbstractManagedObject.
+ * @tparam AccessViaCopies Whether the default access (getters) for this
+ * container provides copies of the objects held, or the actual objects
+ * themselves.
  */
-template <class T>
+template <class T, bool AccessViaCopies>
```

Comment:
I don't like the `bool` parameter, it's not very descriptive. What about this?

```cpp
enum class ManagedContainerAccess {
   Copy,
   Share
};

template<class T, ManagedContainerAccess access> class ManagedContainer;
```

and then possibly shorthands

```cpp
template<class T> using ManagedContainerWithCopyAccess = ManagedContainer<T, ManagedContainerAccess::Copy>;
template<class T> using ManagedContainerWithSharedAccess = ManagedContainer<T, ManagedContainerAccess::Copy>;
```

umm, though these are not really shorter and only add further congitive overhead. So maybe without?

Reply:
Cool, I'll modify to consume an enum, very nice ty.